File: 42586.cpp
#include <string>
#include <vector>
#include <queue>
#include <math.h>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> estimation;
    
    for(int i = 0; i < progresses.size(); i++){
        estimation.push(ceil((100 - progresses[i]) / speeds[i]));
    }
    
    int task = estimation.front();
    int count = 0;
    while(true){
        if(estimation.empty()){
            answer.push_back(count);
            break;
        }
        if(task >= estimation.front()){
            estimation.pop();
            count++;
            continue;
        }
        task = estimation.front();
        answer.push_back(count);
        count = 0;
    }
    return answer;
}

File: 42586.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    var remainingDays = zip(progresses, speeds).map { ceil(Double(100 - $0) / Double($1)) }
    var answer: [Int] = []
    
    while true {
        if remainingDays.isEmpty { break }
        remainingDays = remainingDays.map { $0 - remainingDays[0] }
        var count = 0
        for each in remainingDays {
            if each <= 0 { 
                count += 1
                remainingDays.removeFirst() 
            } else {
                break
            }
        }
        answer.append(count)
    }
    
    return answer
}
